pkg/instance/instance.go

```diff
@@ -525,3 +535,26 @@ func RunnerCmd(prog, fwdAddr, os, arch string, poolIdx, vmIdx int, threaded, new
 	return fmt.Sprintf("%s -addr=%s -os=%s -arch=%s -pool=%d -vm=%d "+
 		"-threaded=%t -new-env=%t", prog, fwdAddr, os, arch, poolIdx, vmIdx, threaded, newEnv)
 }
+
+type BuildSemaphore chan struct{}
+
+func NewBuildSemaphore(count int) BuildSemaphore {
+	return make(chan struct{}, count)
+}
+
+func (s BuildSemaphore) Wait() {
+	s <- struct{}{}
+}
+
+func (s BuildSemaphore) WaitC() chan struct{} {
+	ret := make(chan struct{}, 1)
```

Comment:
If you invert the semaphore with:
```
func NewBuildSemaphore(count int) BuildSemaphore {
	var s BuildSemaphore = make(chan struct{}, count)
	for i := 0; i < count; i++ {
		s.Signal()
	}
}
```
then this can be just `return s`.